core/domain/story_domain.py

```diff
@@ -326,10 +362,23 @@ def validate(self):
             raise utils.ValidationError(
                 'Expected nodes field to be a list, received %s' % self.nodes)
 
-        if not isinstance(self.starting_node_id, basestring):
+        if not isinstance(self.initial_node_id, basestring):
             raise utils.ValidationError(
                 'Expected starting node id to be a string, received %s'
-                % self.starting_node_id)
+                % self.initial_node_id)
+
+        if NODE_ID_PREFIX != self.initial_node_id[:len(NODE_ID_PREFIX)]:
```

Comment:
This is insufficient -- see above. Also, given that this sort of logic is repeated in multiple places I recommend a classmethod along the lines of StoryNode.require_valid_node_id(string_to_test) -- remember the DRY concept. The validation that it is a string should be done outside that so that it's guaranteed that what's passed in is a string.

Reply:
Done, added the function with regex checking and called that everywhere where node is was used.